packages/core/integration-tests/test/bundler.js

```diff
@@ -1214,6 +1214,40 @@ describe('bundler', function () {
     ]);
   });
 
+  it('should support inline constants in async bundles', async () => {
+    await fsFixture(overlayFS, __dirname)`
+    inline-constants-async
+      index.js:
+        import('./async').then(m => console.log(m.value));
+
+      async.js:
+        export const value = 'async value';
+
+      package.json:
+        {
+          "@parcel/transformer-js": {
+            "unstable_inlineConstants": true
+          }
+        }
+
+      yarn.lock:`;
+
+    let b = await bundle(
+      path.join(__dirname, 'inline-constants-async/index.js'),
+      {
+        mode: 'production',
+        defaultTargetOptions: {
+          shouldScopeHoist: true,
+          sourceMaps: false,
+          shouldOptimize: false,
+        },
+        inputFS: overlayFS,
+      },
+    );
+
+    // This will fail when the async bundle does not export it's constant
+    await run(b);
```

Comment:
Does the asset still get inlined in this case? It wouldn't be ideal if you ended up with something like:
```js
import('./async').then(m => console.log('async value'));
```

Reply:
Good question, it doesn't seem to - I assume the static analysis can't track the usage to that level anyway?

From the bundle in the integration test:
```js
(parcelRequire("8Vicm")).then((m)=>console.log(m.value));
```

Reply:
I've added additional verification in the integration test.